cmd/spiffe-aws-assume-role/cli/cli.go

```diff
@@ -10,23 +11,34 @@ import (
 )
 
 type CredentialsCmd struct {
-	Audience       string `required:"" help:"SVID JWT Audience. Must match AWS configuration"`
-	SpiffeID       string `required:"" help:"The SPIFFE ID of this workload"`
-	WorkloadSocket string `optional:"" help:"Path to SPIFFE Workload Socket"`
-	RoleARN        string `required:"" help:"AWS Role ARN to assume"`
-	SessionName    string `optional:"" help:"AWS Session Name"`
-	STSEndpoint    string `optional:"" help:"AWS STS Endpoint variable"`
+	Audience        string        `required help:"SVID JWT Audience. Must match AWS configuration"`
+	SpiffeID        string        `required help:"The SPIFFE ID of this workload"`
+	WorkloadSocket  string        `optional help:"Path to SPIFFE Workload Socket"`
+	RoleARN         string        `required help:"AWS Role ARN to assume"`
+	SessionName     string        `optional help:"AWS Session Name"`
+	STSEndpoint     string        `optional help:"AWS STS Endpoint variable"`
+	SessionDuration time.Duration `optional type:iso8601duration help:"AWS session duration in ISO8601 duration format (e.g. PT5M for five minutes)"`
 }
 
-func (c *CredentialsCmd) Run() error {
+type CliContext struct {
+	JWTSourceProvider credentials.JWTSourceProvider
+	STSProvider       credentials.STSProvider
+}
+
+func (c *CredentialsCmd) Run(context *CliContext) error {
 	spiffeID, err := spiffeid.FromString(c.SpiffeID)
 	if err != nil {
 		return err
 	}
```

Comment:
Building off the previous comment, our CliContext instance is made available to our custom code simply by declaring it as a parameter on our Run method.